File: Employees/src/main/java/com/medecineWebApp/Employees/models/LeaveType.java
package com.medecineWebApp.Employees.models;

import com.medecineWebApp.Configuration.enums.EntityStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;

@Entity
@Table(name = "config_Leave_type")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class LeaveType extends Auditable implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(unique = true, nullable = false)
    private String leaveType;
    @Column(nullable = false)
    private int leaveDays;
    @Enumerated(EnumType.STRING)
    private EntityStatus status;


}

File: Client/frontend/src/app/models/leaveType.ts
import {EntityStatus} from "./Enum/EntityStatus";

export interface ILeaveType {
id?:number,
  leaveType?:string,
  leaveDays?:number,
  status?:EntityStatus,
  description?:string;
}

export class LeaveType implements ILeaveType{
  constructor(
    public id?:number,
    public leaveType?:string,
   public leaveDays?:number,
   public status?:EntityStatus,
   public description?:string,
  ) {

  }
}
